aiodocker/docker.py

```diff
@@ -93,11 +93,33 @@ def __init__(self,
         data = await self._query_json("version")
         return data
 
-    async def pull(self, image, stream=False):
+    async def pull(self, image, auth=None,stream=False):
+
+        headers = {"content-type": "application/json",}
+        if auth:
+            if isinstance(auth, dict) and 'auth' in auth:
+                s = base64.b64decode(auth['auth'])
+                username, pwd = s.split(b':', 1)
+                username = username.decode('utf-8')
+                pwd = pwd.decode('utf-8')
+
+                registry, p, _ = image.partition('/')
+                if not p:
+                    raise ValueError(" image should have registry host")
```

Comment:
Uh-oh. Image names may not have the registry part when we use the default public registry, even with a valid authentication information. In such cases, we need to fill the registry address with a default value (e.g., `registry.docker.io`).

Also, I'm afraid that this implementation only considers login-based authentication, while `X-Registry-Auth` header may be used with token-based authentication, after contacting a separate authentication endpoint.
(see https://docs.docker.com/engine/api/v1.26/#section/Authentication)